### PR TITLE
Implement RFC 2617 - HTTP digest authentication

### PR DESCRIPTION
**Pull request by kiesel**

### CODE REVIEW

File: src/main/php/peer/http/Authorization.class.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public function getPassword() {
+    return $this->password;
+  }
+
+  /**
+   * Set password
+   * 
+   * @param security.SecureString $p
+   */
+  public function setPassword(SecureString $p) {
```
Comment (thekid) [inline src/main/php/peer/http/Authorization.class.php]: You could also use short syntax here, `/** @param security.SecureString $p */`

File: src/main/php/peer/http/Authorization.class.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+abstract class Authorization extends Object {
+  protected $username;
+  protected $password;
+
+  /** @return string */
+  public function getUsername() {
+    return $this->username;
+  }
+
```
Comment (thekid) [inline src/main/php/peer/http/Authorization.class.php]: And if the method only has one line I usually write: 

```php
/** @return string */
public function getUsername() { return $this->username; }
```

File: src/main/php/peer/http/BasicAuthorization.class.php [hunk 3]
```diff
@@ -29,39 +26,20 @@ class BasicAuthorization extends Header {
...
-    parent::__construct('Authorization', 'Basic');
-  }
-
-  /**
-   * Returns the username
-   *
-   * @return  string
-   */    
-  public function getUser() {
-    return $this->user;
```
Comment (thekid) [inline src/main/php/peer/http/BasicAuthorization.class.php]: BC break here, the method above is called `getUsername()`...

**Comment by thekid** [comment]
Here's the example from the `HttpConnection`'s class apidoc:

```php
$c= new HttpConnection('http://xp-framework.net/');
$response= $c->get(
  array('a' => 'b'),
  array(
    new Header('X-Binford', '6100 (more power)'),
    new BasicAuthorization('baz', 'bar'),
    'Cookie' => 'username=fred; lastvisit=2004-01-10'
  )
);
Console::writeLine('Headers: ', $response);
while ($chunk= $response->readData()) {
  // ...
}
```

This wouldn't work anymore AFAIS. Can we restore BC here?

**Comment by thekid** [comment]
:+1:

**Comment by kiesel** [comment]
Superseded by PR #9

**Pull request closed without merging**